bsp/src/org/jetbrains/bsp/project/BspTask.scala

```diff
@@ -76,7 +76,7 @@ class BspTask[T](project: Project,
   }
 
   override def onCancel(): Unit = {
-    resultPromise.failure(new ProcessCanceledException())
+    resultPromise.tryFailure(new ProcessCanceledException())
```

Comment:
@jastice do we need `onCancel` handler at all?

Reply:
I think so, yes. Why do you think not?

Reply:
I noticed the resultPomise is already completed in `BuildToolWindowReporter.actionPerformed`

Reply:
And then the `BspTask.run` completes it as well. Is it possible it's not complited at `onCancel` stage?